### ISSUE TITLE
Fix low accuracy

### ISSUE THREAD

**Issue by Jopyth** [HPI-DeepLearning/ssnn#35]
We probably have an error in one of our layers, or the optimizers?

**Comment by Jopyth** [comment]
Tensorflow reports:
```
Epoch: 0001 cost=247.178556213
Accuracy: 0.8528
```
for a MLP model with `learning_rate = 0.001` and 256 hidden units in each fully connected layer (SGD)/

**Comment by Jopyth** [comment]
Sample reading seems to be wrong.

**Comment by Jopyth** [comment]
Found it, in this [line](https://github.com/HPI-DeepLearning/length/blob/master/length/data_sets/mnist_like.py#L85) we are reading a `np.int8` instead of `np.uint8`.
